reportportal_client/service.py

```diff
@@ -249,31 +251,53 @@ def finish_launch(self, end_time, status=None, **kwargs):
         logger.debug("finish_launch - ID: %s", self.launch_id)
         return _get_msg(r)
 
-    def get_launch_info(self):
+    def get_launch_info(self, max_retries=5):
         """Get the current launch information.
 
+        Perform "max_retries" attempts to get current launch information
+        with 0.5 second sleep between them.
+
+        :param int max_retries: Number of retries to get launch information.
         :return dict: launch information
         """
+        if self.launch_id is None:
+            raise LaunchExistsError(
```

Comment:
We should not raise anything from those functions. They provide additional functionality and should not impact reporting.

Reply:
No exceptions? It is a Python! :) Ok, will do follow the project style.  In this case, I have to return an empty dict since we got no information from server. I prefer to fail fast approach. If we remove the exception here it will fail with Key error later.

Reply:
Every time you create an exception, you will have to keep in mind that it needs to be handled by everyone who will use that piece of code. It makes your program harder to maintain and debug, especially if threads are in use.

For this particular functionality, I'd polling for the ID 5 times and then log message like "Failed to fetch launch ID from the API" with error or warning level. Key error exception should not break the execution of the "reporting" as well.

Reply:
Thanks for the explanation of the project`s style. It is just a fail-fast vs fail-safe approach.  In many places exception are great and allows to write more readable and pythonic code. I have changed the code per your comment. 

If we do not rais an exception in case of an error - we have to return something. Added return all launches URL, in this case, to make the behavior more user friendly.